biodatasets/n2c2_2008/n2c2_2008.py

```diff
@@ -0,0 +1,366 @@
+        ) as t2, open(data_dir / "obesity_standoff_annotations_training_addendum2.xml") as t3, open(
+            data_dir / "obesity_standoff_annotations_training_addendum3.xml"
+        ) as t4:
+            train1 = t1.read().strip()
+            train2 = t2.read().strip()
+            train3 = t3.read().strip()
+            train4 = t4.read().strip()
+        annotation_files = (train1, train2, train3, train4)
+    elif partition == "test":
+        with open(data_dir / "obesity_standoff_annotations_test.xml") as t1:
+            test1 = t1.read().strip()
+        annotation_files = (test1,)
+
+    for file in annotation_files:
+        root = et.fromstring(file)
+        for diseases_annotation in root.findall("./diseases"):
+
+            annotation_source = diseases_annotation.attrib["source"]
+            assert isinstance(annotation_source, str)
+            for disease in diseases_annotation.findall("./disease"):
+                disease_name = disease.attrib["name"]
+                all_diseases.add(disease_name)
+                for annotation in disease.findall("./doc"):
+                    doc_id = annotation.attrib["id"]
+                    if not annotation_source in documents[doc_id]:
+                        documents[doc_id][annotation_source] = {}
+                    assert doc_id in documents
+                    judgment = annotation.attrib["judgment"]
+                    documents[doc_id][annotation_source][disease_name] = judgment
+    return [
+        {
+            "document_id": id,
+            "text": documents[id]["text"],
+            "textual": _map_labels(documents[id], "textual"),
+            "intuitive": _map_labels(documents[id], "intuitive"),
+        }
+        for id in documents
+    ]
+
+
+class N2C22008ObesityDataset(datasets.GeneratorBasedBuilder):
+    """n2c2 2008 obesity and comorbidities recognition task"""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    BIGBIO_VERSION = datasets.Version(_BIGBIO_VERSION)
+
+    BUILDER_CONFIGS = [
+        BigBioConfig(
+            name="n2c2_2008_source",
+            version=SOURCE_VERSION,
+            description="n2c2_2008 source schema",
+            schema="source",
+            subset_id="n2c2_2008",
+        ),
+        BigBioConfig(
+            name="n2c2_2008_bigbio_text",
+            version=BIGBIO_VERSION,
+            description="n2c2_2008 BigBio schema",
+            schema="bigbio_text",
+            subset_id="n2c2_2008",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "n2c2_2008_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "document_id": datasets.Value("string"),
+                    "text": datasets.Value("string"),
+                    "textual": [datasets.Value("string")],
+                    "intuitive": [datasets.Value("string")],
```

Comment:
does it make sense to use `ClassLabel` instead of `Value("string")` here for `textual` and/or `intuitive` ?

Reply:
or alternatively ... is there a more natural set of features for the source schema given we want to preserve as much information as possible? 
```
"textual": [
    "disease_name": ClassLabel,
    "judgement": ClassLabel,
]
```

or 

```
"disease_set": [
    "source": ClassLabel(names=["textual", "intuitive"]),
    "name": ClassLabel(names=["Asthma", ...]),
    "judgement": ClassLabel(names=["Y", "N", "Q", "U"]) # or "present", "absent", ...
]
```

(not sure if it makes sense to use ClassLabel to indicate an enum that is not (by itself) a label, but maybe? )